molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+---
+
+- name: Create
+  hosts: localhost
+  gather_facts: false
+  no_log: "{{ molecule_no_log }}"
+  vars:
+    molecule_scenario_name: "{{ lookup('env', 'MOLECULE_SCENARIO_NAME') }}"
+    user_home_directory: "{{ lookup('env', 'HOME') }}"
+    molecule_dns_suffix: "molecule.lab"
+    molecule_network_name: "molecule-libvirt"
+    molecule_ssh_user: "molecule"
+    molecule_private_key_path: "{{ lookup('env', 'MOLECULE_EPHEMERAL_DIRECTORY') }}/id_ssh_rsa"
+    libvirt_default_image_path: "{{ user_home_directory}}/.local/molecule/{{ molecule_scenario_name }}/vms"
+    libvirt_vm_arch: "x86_64"
+    libvirt_uri: "qemu:///system"
+    default_image_url: "https://mirror.csclub.uwaterloo.ca/fedora/linux/releases/33/Cloud/x86_64/images/Fedora-Cloud-Base-33-1.2.x86_64.qcow2"
+    default_disk_size: "15G"
+    default_memory_size: "1"
+    default_vcpu: "1"
+    default_ssh_port: "22"
+    mac: "52:54:00:ab:cd:"  # first 5 digits
+    temp_folder: "/tmp"
+
+  tasks:
+
+    - name: Ensure default vm folder exists
+      file:
+        path: "{{ libvirt_default_image_path }}"
+        state: directory
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: Allow qemu process read qcow2 images in home directory
+      acl:
+        path: "{{ user_home_directory }}"
+        entity: qemu
+        etype: user
+        permissions: x
+        state: present
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: "Create network {{ molecule_network_name }}"
+      virt_net:
+        command: define
+        name: "{{ molecule_network_name }}"
+        xml: "{{ lookup('template', 'network/network.xml.j2') }}"
+        uri: "{{ qemu_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+      vars:
+        qemu_uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+
+    - name: "Start network {{ molecule_network_name }}"
+      virt_net:
+        command: start
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+
+    - name: Get qcow2 image
+      get_url:
+        url: "{{ image_url | default(default_image_url) }}"
+        dest: "{{ libvirt_default_image_path }}/{{ item.name }}.qcow2"
+        mode: '0660'
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
with_items has been changed for loop.